internal/util/util.go

```diff
@@ -37,23 +39,76 @@ func BundleInstanceProvisionerFilter(provisionerClassName string) predicate.Pred
 
 func MapBundleToBundleInstanceHandler(cl client.Client, log logr.Logger) handler.MapFunc {
 	return func(object client.Object) []reconcile.Request {
-		b := object.(*rukpakv1alpha1.Bundle)
+		b, ok := object.(*rukpakv1alpha1.Bundle)
+		if !ok {
+			return nil
+		}
 		bundleInstances := &rukpakv1alpha1.BundleInstanceList{}
-		var requests []reconcile.Request
 		if err := cl.List(context.Background(), bundleInstances); err != nil {
 			log.WithName("mapBundleToBundleInstanceHandler").Error(err, "list bundles")
-			return requests
+			return nil
 		}
+		var requests []reconcile.Request
 		for _, bi := range bundleInstances.Items {
 			bi := bi
-			if bi.Spec.BundleName == b.Name {
-				requests = append(requests, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&bi)})
+			for _, ref := range b.GetOwnerReferences() {
+				if ref.Name == bi.GetName() {
+					requests = append(requests, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&bi)})
+				}
```

Comment:
Should we go by label selector instead of owner reference? Or maybe both?

Reply:
I can update to use a label selector implementation.

Reply:
I might have deleted my focus for the day but I wonder whether we should be injecting the core.rukpak.io/owner-kind label on the generated Bundle resource so we only requeue owning BundleInstance resources vs. any BundleInstance that has matching labels?

Reply:
I have to reboot my memory on that, to be honest. IIRC those labels we're mainly used for `SelectorsByObject` to limit what the informers watch/cache.

Reply:
I think deployments/replica sets can technically adopt overlapping sets of pods. I have no idea what that would mean or whether there's an actual use case for that.

Reply:
Here's what deployment does for listing its replica sets. Looks like it uses the label selector and then furhter filters by the owner ref.

Reply:
We hashed this out in the [rukpak-dev slack channel](https://github.com/operator-framework/rukpak/pull/293/commits/d7bb57295d7037817838f6222d05cf28f526ec52).

Update the implementation to use label selectors in https://github.com/operator-framework/rukpak/pull/293/commits/d7bb57295d7037817838f6222d05cf28f526ec52.